common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -140,19 +144,41 @@ public Random current() {
-                    "Your platform does not provide complete low-level API for accessing direct buffers reliably. " +
-                    "Unless explicitly requested, heap buffer will always be preferred to avoid potential system " +
-                    "instability.");
-        }
-
-        // Here is how the system property is used:
-        //
-        // * <  0  - Don't use cleaner, and inherit max direct memory from java. In this case the
-        //           "practical max direct memory" would be 2 * max memory as defined by the JDK.
-        // * == 0  - Use cleaner, Netty will not enforce max memory, and instead will defer to JDK.
-        // * >  0  - Don't use cleaner. This will limit Netty's total direct memory
-        //           (note: that JDK's direct memory limit is independent of this).
-        long maxDirectMemory = SystemPropertyUtil.getLong(JAVA_SYS_PROP_IO_NETTY_MAX_DIRECT_MEMORY, -1);
+              "Your platform does not provide complete low-level API for accessing direct buffers reliably. " +
+                "Unless explicitly requested, heap buffer will always be preferred to avoid potential system " +
+                "instability.");
+        }
+
+        // Make sure that 3rd party library had the chance to set the max direct memory prior to getting in here.
+        try {
+            PlatformDependentCompanion.checkSetMaxDirectMemoryWasCalled();
+        } catch (IllegalAccessException e) {
+            logger.error("To avoid this exception please either let Netty use the default value for " +
+              "the max direct memory by calling PlatformDependentCompanion.setUseDefaultMaxDirectMemory() or call " +
+              "PlatformDependentCompanion.setMaxDirectMemory(long m) to tell Netty the max direct memory to use here.");
+            throwException(e);
+        }
+
+        long maxDirectMemory = PlatformDependentCompanion.getMaxDirectMemory();
+        if (maxDirectMemory < 0) {
+            // Here is how the system property is used:
+            //
+            // * <  0  - Don't use cleaner, and inherit max direct memory from java. In this case the
+            //           "practical max direct memory" would be 2 * max memory as defined by the JDK.
+            // * == 0  - Use cleaner, Netty will not enforce max memory, and instead will defer to JDK.
+            // * >  0  - Don't use cleaner. This will limit Netty's total direct memory
+            //           (note: that JDK's direct memory limit is independent of this).
+            maxDirectMemory = SystemPropertyUtil.getLong(JAVA_SYS_PROP_IO_NETTY_MAX_DIRECT_MEMORY, -1);
+
+            if (maxDirectMemory >= 0) {
+                logger.warn("Using java system property '{}' is deprecated and will be removed in the future in favor" +
+                    " of using one of the setter methods from the class '{}'" ,
+                  JAVA_SYS_PROP_IO_NETTY_MAX_DIRECT_MEMORY, PlatformDependentCompanion.class.getName());
+            }
```

Comment:
This check should be done in Federate plugin side, not netty imho.

Reply:
That makes sense as Federate deprecates its usage regardless of what Netty does (ref.: https://github.com/sirensolutions/netty/pull/12/files#r539184605)